blocks/rich-text/index.js

```diff
@@ -612,8 +655,7 @@ export class RichText extends Component {
 			const afterFragment = afterRange.extractContents();
 
 			const beforeElement = nodeListToReact( beforeFragment.childNodes, createTinyMCEElement );
-			const afterElement = isLinkBoundary( afterFragment ) ? [] : nodeListToReact( afterFragment.childNodes, createTinyMCEElement );
-
+			const afterElement = nodeListToReact( filterEmptyNodes( afterFragment.childNodes ), createTinyMCEElement );
```

Comment:
Could the same not happen on `beforeElement`?

Reply:
The issue I'm seeing is that if the empty elements (`br`) are in the middle of the after content, they are also removed from the newly created block.

Reply:
In theory, this issue could happen in `beforeElement` when you trigger Enter at the beginning of a paragraph block but in my testings the block is considered Empty properly.